### PR TITLE
Update cobra to v1.4.0

### LINKED ISSUE #1166: Update Cobra to version 1.3.0
**What would you like to be added**:
Upgrade cobra from version 1.2.1 to 1.3.0

**Why is this needed**:
* Enables displaying command descriptions in bash completion output like it does for zsh and fish completions currently (bug fixed in cobra 1.3.0 to make this usable https://github.com/spf13/cobra/pull/1509)
* Improves flag parsing capabilities which will help with plugin command completions (https://github.com/kubernetes/kubernetes/pull/105867#issuecomment-995264504)
* Several other minor fixes and enhancements in Cobra 1.3.0 related to completions

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

Retread of https://github.com/kubernetes/kubernetes/pull/107628 on 1.4.0

/kind cleanup
/kind feature

#### What this PR does / why we need it:

Updates to cobra 1.4.0, picks up features highlighted in https://github.com/kubernetes/kubernetes/pull/107628, and drops the viper dependency 🎉 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubectl/issues/1166

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/sig cli
cc @eddiezane @brianpursley @dims

### CODE REVIEW

File: hack/unwanted-dependencies.json [hunk 1]
```diff
@@ -1,30 +1,30 @@
 { 
   "spec": {
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:2]: diff here is strictly additive, I just sorted the list as well

File: hack/unwanted-dependencies.json [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+      "rsc.io/sampler": "refer to #102833"
     }
   },
   "status": {
     "references": [
       "github.com/go-kit/kit",
-      "github.com/hashicorp/golang-lru",
-      "github.com/hashicorp/hcl",
-      "github.com/json-iterator/go",
-      "github.com/spf13/viper"
```
Comment (liggitt) [inline hack/unwanted-dependencies.json:27]: three of these dropped out of our deps list as a result of this :)

**Comment by liggitt** [comment]
dependency stats:

```diff
{
-	"directDependencies": 160,
-	"transitiveDependencies": 307,
-	"totalDependencies": 367,
-	"maxDepthOfDependencies": 30
+	"directDependencies": 161,
+	"transitiveDependencies": 262,
+	"totalDependencies": 323,
+	"maxDepthOfDependencies": 29
 }
```

transitive: 307 → 262 (-14.6%)

total: 367 → 323 (-11.9%)

**Comment by dims** [comment]
looks like the only other bump was `go-md2man/v2` from `v2.0.0` to `v2.0.1`

Win! Win!

/approve
/lgtm

**Comment by liggitt** [comment]
and `github.com/russross/blackfriday/v2 => github.com/russross/blackfriday/v2 v2.1.0`, which was a pre-existing dep

**Pull request merged**